src/components/breadcrumb-bar/get-breadcrumb-links.ts

```diff
@@ -0,0 +1,128 @@
+import { BreadcrumbLink } from '.'
+
+/**
+ * Given a basePath, navData, and array of path parameters,
+ * Return an array of breadcrumb links leading to this page.
+ *
+ * Note that the returned array does not include links beyond
+ * the basePath. So, in many use cases, consumers will likely
+ * want to prefix this array to provide a more complete
+ * list of breadcrumb links.
+ */
+function getBreadcrumbLinks({
```

Comment:
There's some complexity here due to the flexibility in our `navData` structures. Namely:
- Some breadcrumb parts correspond to `{ title, path }` nodes with useful titles. These are easy.
- Some breadcrumb parts correspond to "index" nodes, which often have the title "Overview". If we naively match these nodes and use their `title` values, we end up with breadcrumbs like `Waypoint > Docs > Overview > Overview > Helm`. So we implement some logic to detect these "index" nodes, and instead use the parent category's title, giving us something like `Waypoint > Docs > Introduction > Waypoint vs. Other Software > Helm`
- Some breadcrumb parts correspond to index-less categories. An example of an index-less category is Waypoint's `kubernetes` category - [/docs/kubernetes/install](https://www.waypointproject.io/docs/kubernetes/install) is a valid page, but [/docs/kubernetes](https://www.waypointproject.io/docs/kubernetes) is a 404 (as `Kubernetes` is a category with no index page). In these cases, we want to show the index-less category's title in the breadcrumb, but not have that breadcrumb part link anywhere (as otherwise, for the `/docs/kubernetes/install` breadcrumb we end up with `Waypoint > Docs > Install`, which is misleading).